Common/IO/FileDemand.cs

```diff
@@ -0,0 +1,188 @@
+                case MatchKind.Name:
+                    return MatchContent.Equals(fileName, StringComparison.InvariantCultureIgnoreCase);
+                case MatchKind.Prefix:
+                    return fileName.StartsWith(MatchContent, StringComparison.InvariantCultureIgnoreCase);
+                case MatchKind.Contains:
+                    return fileName.ToLower().Contains(MatchContent.ToLower());
+                case MatchKind.Suffix:
+                    return fileName.EndsWith(MatchContent, StringComparison.InvariantCultureIgnoreCase);
+                case MatchKind.Pattern:
+                    return ToRegex().IsMatch(fileName);
+                default:
+                    throw new NotSupportedException();
+            }
+        }
+
+        public override string ToString()
+        {
+            switch (MatchLevel)
+            {
+                case MatchKind.FullPath:
+                case MatchKind.Name:
+                    return $"a file named '{MatchContent}'";
+                case MatchKind.Prefix:
+                    return $"a file starting with '{MatchContent}'";
+                case MatchKind.Contains:
+                    return $"a file with '{MatchContent}' in the name";
+                case MatchKind.Suffix:
+                    return $"a file ending with '{MatchContent}'";
+                case MatchKind.Pattern:
+                    return $"a file matching the pattern '{GetPrintablePattern()}'";
+                default:
+                    throw new NotSupportedException();
+            }
+        }
+
+        public Regex ToRegex()
+        {
+            switch (MatchLevel)
+            {
+                case MatchKind.Pattern:
+                    return new Regex(MatchContent, RegexOptions.IgnoreCase);
+                default:
+                    return null;
+            }
+        }
+
+        private string GetPrintablePattern()
+        {
+            string prettyPattern = MatchContent
+                                    .Replace(@"\-", "-")
+                                    .Replace(@"\.",".")
+                                    .Replace(@"[0-9]", "#");
+
+            if (prettyPattern.StartsWith("^"))
+            {
+                prettyPattern = prettyPattern.Substring(1);
+            }
+
+            if (prettyPattern.EndsWith("$"))
+            {
+                prettyPattern = prettyPattern.Substring(0, prettyPattern.Length - 1);
+            }
+
+            const string matchAnyOrNone = ".*?";
+            const string matchAnyAtLeastOnce = ".+?";
+            if (prettyPattern.Contains(matchAnyAtLeastOnce) || prettyPattern.Contains(matchAnyOrNone))
+            {
+                prettyPattern = prettyPattern.Replace(matchAnyAtLeastOnce, "*").Replace(matchAnyOrNone, "*");
+            }
+
+            prettyPattern = Regex.Replace(prettyPattern, @"\(.+?\|.+?\)", "*");
+            prettyPattern = Regex.Replace(prettyPattern, @"\\d\{(\d,)?\d\}", "*");
```

Comment:
I don't think the compiler would care, but you are chaining the .Replace on line 127, but then not chaining on line 149 just being consistent might be your intent.

Reply:
line 127 is using instance method, but we can't chain here because there are static methods.